crates/autopilot/src/domain/settlement/transaction/mod.rs

```diff
@@ -9,6 +9,30 @@ use {
 
 mod tokenized;
 
+/// The following trait allows to implement custom solver authentication logic
+/// for transactions.
+#[async_trait::async_trait]
+pub trait TransactionAuthenticator {
+    /// Determines whether the provided address is an authenticated solver.
+    async fn is_solver(&self, prospective_solver: ethcontract::Address) -> Result<bool, Error>;
```

Comment:
Since this is in the domain we should use the domain new types.
```suggestion
    async fn is_solver(&self, prospective_solver: eth::Address) -> Result<bool, Error>;
```